core/txpool/blobpool/blobpool.go

```diff
@@ -0,0 +1,1461 @@
+			log.Error("Failed to delete replaced transaction", "id", prev.id, "err", err)
+		}
+		// Update the transaction index
+		p.index[from][offset] = meta
+		p.spent[from] = new(uint256.Int).Sub(p.spent[from], prev.costCap)
+		p.spent[from] = new(uint256.Int).Add(p.spent[from], meta.costCap)
+
+		delete(p.lookup, prev.hash)
+		p.lookup[meta.hash] = meta.id
+		p.stored += uint64(meta.size) - uint64(prev.size)
+	} else {
+		// Transaction extends previously scheduled ones
+		p.index[from] = append(p.index[from], meta)
+		if _, ok := p.spent[from]; !ok {
+			p.spent[from] = new(uint256.Int)
+			newacc = true
+		}
+		p.spent[from] = new(uint256.Int).Add(p.spent[from], meta.costCap)
+		p.lookup[meta.hash] = meta.id
+		p.stored += uint64(meta.size)
+	}
+	// Recompute the rolling eviction fields. In case of a replacement, this will
+	// recompute all subsequent fields. In case of an append, this will only do
+	// the fresh calculation.
+	var (
+		txs = p.index[from]
+
+		oldEvictionExecFeeJumps = txs[len(txs)-1].evictionExecFeeJumps
+		oldEvictionBlobFeeJumps = txs[len(txs)-1].evictionBlobFeeJumps
+	)
+	for i := offset; i < len(txs); i++ {
+		// The first transaction will always use itself
+		if i == 0 {
+			txs[0].evictionExecTip = txs[0].execTipCap
+			txs[0].evictionExecFeeJumps = txs[0].basefeeJumps
+			txs[0].evictionBlobFeeJumps = txs[0].blobfeeJumps
+
+			continue
+		}
+		// Subsequent transactions will use a rolling calculation
+		txs[i].evictionExecTip = txs[i-1].evictionExecTip
+		if txs[i].evictionExecTip.Cmp(txs[i].execTipCap) > 0 {
+			txs[i].evictionExecTip = txs[i].execTipCap
+		}
+		txs[i].evictionExecFeeJumps = txs[i-1].evictionExecFeeJumps
+		if txs[i].evictionExecFeeJumps > txs[i].basefeeJumps {
+			txs[i].evictionExecFeeJumps = txs[i].basefeeJumps
+		}
+		txs[i].evictionBlobFeeJumps = txs[i-1].evictionBlobFeeJumps
+		if txs[i].evictionBlobFeeJumps > txs[i].blobfeeJumps {
+			txs[i].evictionBlobFeeJumps = txs[i].blobfeeJumps
+		}
+	}
+	// Update the eviction heap with the new information:
+	//   - If the transaction is from a new account, add it to the heap
+	//   - If the account had a singleton tx replaced, update the heap (new price caps)
+	//   - If the account had it's last tx updated or appended to, update heap iff lower than prev
+	//   - If the account had an internal tx updated, compare old tail price caps with new ones
+	switch {
+	case newacc:
+		heap.Push(p.evict, from)
+
+	case len(txs) == 1: // 1 tx and not a new acc, must be replacement
+		heap.Fix(p.evict, p.evict.index[from])
+
+	case offset == len(txs)-1: // either append or last tx replacement
+		evictionExecFeeDiff := txs[offset-1].evictionExecFeeJumps - txs[offset].evictionExecFeeJumps
+		evictionBlobFeeDiff := txs[offset-1].evictionBlobFeeJumps - txs[offset].evictionBlobFeeJumps
+
+		if evictionExecFeeDiff > 0.001 || evictionBlobFeeDiff > 0.001 { // no need for math.Abs, monotonic decreasing
```

Comment:
We use `0.001` here and `0.01` somewhere else. Better to have an aligned threshold.

Reply:
LES used 0.01, seems a bit too granular for me.